SalesforceCommerceCloud/pwa-kit
# @W-16795956 - Implement "Check Email" page

# Description
In the passwordless login flow, a passwordless token is sent to the shopper over email. Once this email is sent, users will be redirected to a "Check Your Email" page.

⚠️This PR only implements the UI⚠️

Figma prototype [here](https://www.figma.com/proto/80M7MTsZT6Ipu8xAGi6qUK/%5B254%5D-Passwordless-and-Social-Login?page-id=&node-id=7238-43549&node-type=canvas&viewport=1300%2C-1369%2C0.25&t=y1BWmAc1rMMzIloo-1&scaling=min-zoom&content-scaling=fixed&starting-point-node-id=7238%3A42847&show-proto-sidebar=1)



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- Add new `PasswordlessEmailConfirmation` component that includes a `Resend Link` button
  - `app/component/email-confirmation`

# How to Test-Drive This PR

- Checkout this branch and start retail app locally

To see the Check Email modal:
- Click on the **Login** person icon on the top right
- **Enter an email**
- Click the **Continue Securely** button
- Verify that you see the **Check Email** modal with a close icon (⚠️Verify that you cannot click the link if there is **no** email inputted⚠️)
<img width="600" alt="Screenshot 2024-11-04 at 10 11 44 AM" src="https://github.com/user-attachments/assets/7f09af34-e833-4d69-bf3d-b5b757a15b72">

To see the Check Email page:
- Navigate to the Login page by going to localhost:3000/login
- **Enter an email**
- Click the **Continue Securely** button
- Verify that you see the **Check Email** page (⚠️Verify that you cannot click the link if there is **no** email inputted⚠️)
<img width="600" alt="Screenshot 2024-11-04 at 10 11 28 AM" src="https://github.com/user-attachments/assets/1d20bf58-f256-4996-b42a-15b50fc4825e">


# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
